relay.go

```diff
@@ -552,6 +552,15 @@ func (r *Relayer) handleNonCallReq(f *Frame) error {
 		return nil
 	}
 
+	if f.messageType() == messageTypeCallRes {
```

Comment:
since we hae another `messageType()` check below, can we use a switch for both these cases?